viewing_party/party.py

```diff
@@ -1,23 +1,269 @@
+    output: dictionary containing the 3 input parameters
+    '''
+
+    # Check to see if the 3 parameters (title, genre, and rating) are truthy.
+    # If so, return a dictionary named movie_details containing these 3 attributes as key-value pairs.
+    # If all 3 parameters are falsy, return None.
+    
+    if title and genre and rating:
+        movie_details = {'title': title, 'genre': genre, 'rating': rating}
+        return movie_details
+    else:
+        return None
+    
+
+
+def add_to_watched(user_data, movie):
+
+    '''
+    input: user_data-dictionary with list of dictionaries & movie dictionary with title of movie watched
+    ouput: modified user_data with movie added to 'watched' list
+    '''
+
+    # Add new movie (dictionary) as a value (list) associated to the key, "watched" within the user_data dictionary. 
+
+    user_data['watched'].append(movie)
+    return user_data
+
+
+
+def add_to_watchlist(user_data, movie):
+
+    '''
+    input: user_data-dictionary with list of dictionaries & movie- dict. with title of movie user wants to watch
+    ouput: modified user_data with movie added to 'watchlist'
+    '''
+
+    # Add new movie (dictionary) as a value (list) associated to the key, "watchlist" within the user_data dictionary. 
+
+    user_data['watchlist'].append(movie)
+    return user_data
+
+
+
+def watch_movie(user_data, title):
+
+    '''
+    input: dictionary user_data, string movie user has watched-title
+    ouput: updated user_data
+    '''
+
+    # Loop through each movie within the "watchlist" nested data structure to see if the title (parameter) exists.
+    # If so, remove that movie from the "watchlist" and append it to the "watched" list of user_data.
+
+    for movies in user_data['watchlist']:
+        if movies['title'] == title:
+            user_data["watchlist"].remove(movies)
+            user_data["watched"].append(movies)
+    return user_data
+    
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+
+    '''
+    input: takes in one parameter in the form of a dictionary named user_data.
+    output: returns the average rating (float) of all the movies within the watched list value in user_data.
+    '''
+
+    # Return 0.0 if the average rating of all the movies int he "watched" list is empty.
+
+    if len(user_data["watched"]) == 0:
+        return 0.0
+
+    # Calculate the average rating of all movies within the "watched" list by creating an
+    # empty dictionary named "ratings", then looping through each movie in the "watched" list
+    # to retrieve and append the rating value of each movie to the "ratings" list, and performing statistics.mean()
+
+    ratings = []
+    for movie in user_data["watched"]:
+        ratings.append(movie["rating"])
+    return statistics.mean(ratings)

```

Comment:
NIIIIICE I love it! By using the `statics` package, this function reads so cleanly.